From 833b892b0654611bf84d18dcb932535b33d94601 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Tue, 9 Apr 2024 08:43:38 +0000
Subject: [PATCH 08/10] bnxt_en: correct the path for auxiliary_bus.h

auxiliary_bus.h is in /usr/src/linux-headers-6.6.0-1-common, so correct
the path for auxiliary_bus.h to fix the build issue.

bnxt_re-227.0.130.0/main.c:1032:40: error: implicit declaration of\
 function auxiliary_get_drvdata [-Werror=implicit-function-declaration]
 1032 |  struct bnxt_re_en_dev_info *en_info = auxiliary_get_drvdata(handle);
      |                                        ^~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 bnxt_en-1.10.2-227.0.130.0/Makefile             | 12 +++++++++---
 bnxt_en-1.10.2-227.0.130.0/bnxt_auxbus_compat.h |  4 ----
 bnxt_re-227.0.130.0/Makefile                    | 11 ++++++++---
 3 files changed, 17 insertions(+), 10 deletions(-)

diff --git a/bnxt_en-1.10.2-227.0.130.0/Makefile b/bnxt_en-1.10.2-227.0.130.0/Makefile
index 2331171..e229cc9 100644
--- a/bnxt_en-1.10.2-227.0.130.0/Makefile
+++ b/bnxt_en-1.10.2-227.0.130.0/Makefile
@@ -1196,7 +1196,13 @@ ifneq ($(shell grep -so "struct netlink_ext_ack" $(LINUXSRC)/include/linux/netli
   DISTRO_CFLAG += -DHAVE_NETLINK_EXT_ACK
 endif
 
-ifneq ($(shell ls $(LINUXSRC)/include/linux/auxiliary_bus.h > /dev/null 2>&1 && echo auxiliary_driver),)
+ifeq (${KSRC_COMMON},)
+KSRC_AUX := ${LINUXSRC}
+else
+KSRC_AUX := ${KSRC_COMMON}
+endif
+
+ifneq ($(shell ls $(KSRC_AUX)/include/linux/auxiliary_bus.h > /dev/null 2>&1 && echo auxiliary_driver),)
   DISTRO_CFLAG += -DHAVE_AUXILIARY_DRIVER
 endif
 
@@ -1208,11 +1214,11 @@ ifneq ($(shell grep -o "struct auxiliary_device_id" $(LINUXSRC)/include/linux/mo
   DISTRO_CFLAG += -DHAVE_AUX_DEVICE_ID
 endif
 
-ifneq ($(shell grep -so "auxiliary_set_drvdata" $(LINUXSRC)/include/linux/auxiliary_bus.h),)
+ifneq ($(shell grep -so "auxiliary_set_drvdata" $(KSRC_AUX)/include/linux/auxiliary_bus.h),)
   DISTRO_CFLAG += -DHAVE_AUX_SET_DRVDATA
 endif
 
-ifneq ($(shell grep -so "auxiliary_get_drvdata" $(LINUXSRC)/include/linux/auxiliary_bus.h),)
+ifneq ($(shell grep -so "auxiliary_get_drvdata" $(KSRC_AUX)/include/linux/auxiliary_bus.h),)
   DISTRO_CFLAG += -DHAVE_AUX_GET_DRVDATA
 endif
 
diff --git a/bnxt_en-1.10.2-227.0.130.0/bnxt_auxbus_compat.h b/bnxt_en-1.10.2-227.0.130.0/bnxt_auxbus_compat.h
index 9496e56..d4b2653 100644
--- a/bnxt_en-1.10.2-227.0.130.0/bnxt_auxbus_compat.h
+++ b/bnxt_en-1.10.2-227.0.130.0/bnxt_auxbus_compat.h
@@ -10,10 +10,6 @@
 #ifndef _BNXT_AUXILIARY_COMPAT_H_
 #define _BNXT_AUXILIARY_COMPAT_H_
 
-#if !defined(CONFIG_AUXILIARY_BUS)
-#undef HAVE_AUXILIARY_DRIVER
-#endif
-
 #ifdef HAVE_AUXILIARY_DRIVER
 #include <linux/auxiliary_bus.h>
 #endif
diff --git a/bnxt_re-227.0.130.0/Makefile b/bnxt_re-227.0.130.0/Makefile
index 600698c..21048b9 100644
--- a/bnxt_re-227.0.130.0/Makefile
+++ b/bnxt_re-227.0.130.0/Makefile
@@ -910,17 +910,22 @@ ifneq ($(shell grep -o "struct auxiliary_device_id" $(LINUXSRC)/include/linux/mo
   DISTRO_CFLAG += -DHAVE_AUX_DEVICE_ID
 endif
 
-ifneq ($(shell ls $(LINUXSRC)/include/linux/auxiliary_bus.h > /dev/null 2>&1 && echo auxiliary_driver),)
+ifeq (${KSRC_COMMON},)
+KSRC_AUX := ${LINUXSRC}
+else
+KSRC_AUX := ${KSRC_COMMON}
+endif
+ifneq ($(shell ls $(KSRC_AUX)/include/linux/auxiliary_bus.h > /dev/null 2>&1 && echo auxiliary_driver),)
   ifneq ($(CONFIG_AUXILIARY_BUS),)
     DISTRO_CFLAG += -DHAVE_AUXILIARY_DRIVER
   endif
 endif
 
-ifneq ($(shell grep -so "auxiliary_set_drvdata" $(LINUXSRC)/include/linux/auxiliary_bus.h),)
+ifneq ($(shell grep -so "auxiliary_set_drvdata" $(KSRC_AUX)/include/linux/auxiliary_bus.h),)
   DISTRO_CFLAG += -DHAVE_AUX_SET_DRVDATA
 endif
 
-ifneq ($(shell grep -so "auxiliary_get_drvdata" $(LINUXSRC)/include/linux/auxiliary_bus.h),)
+ifneq ($(shell grep -so "auxiliary_get_drvdata" $(KSRC_AUX)/include/linux/auxiliary_bus.h),)
   DISTRO_CFLAG += -DHAVE_AUX_GET_DRVDATA
 endif
 
-- 
2.43.0

